File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        if(nums.size() == 0 ) return 0;
        int count = 1, res = 1;
        for(int i = 0;i<nums.size();i++){
            if(nums[i] == nums[i+1]){
                continue;
            }
            else if(nums[i]+1 == nums[i+1]){
                count++;
                res = max(res, count);
            }
            else{
                count = 1;
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if not s:
            return 0

        h = [s[0]] 
        l,r = 0,1
        res = 1

        while r<len(s):
            while s[r] in h:
                h.remove(s[l])
                l+=1
            res = max(res, (r-l+1))
            h.append(s[r])
            r+=1
        
        return res